### PR TITLE
Remove usages of 'Foo' from our codebase.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
It doesn't pass policheck.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @jaredpar .  I preserved casing as that's very significant for many usages of these variables.  i.e. if we had "fOo" it was changed to "gOo".  

I expect some failures after the first wave.  Specifically, for things like completion or matching where we'll have to change what is typed to match.  i.e. we'll likely have to change an "f" that was typed in a test to a "g".

File: src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs [hunk 1]
```diff
@@ -1063,7 +1063,7 @@ public IAliasSymbol GetAliasInfo(IdentifierNameSyntax nameSyntax, CancellationTo
             if (!CanGetSemanticInfo(nameSyntax))
                 return null;
 
-            SymbolInfo info = GetSymbolInfoWorker(nameSyntax, SymbolInfoOptions.PreferTypeToConstructors | SymbolInfoOptions.PreserveAliases, cancellationToken);
+            SymbolInfo info = GetSymbolInfoWorker(nameSyntax, SymbolIngoOptions.PreferTypeToConstructors | SymbolIngoOptions.PreserveAliases, cancellationToken);
```
Comment (jamesqo) [inline src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs]: This seems unintentional

**Comment by CyrusNajmabadi** [comment]
@sharwell Are you commenting on the right PR?  This PR is about removing usages of Foo in our codebase due to it not passing Policheck.

**Comment by CyrusNajmabadi** [comment]
> The new name in each case is more meaningful in the context of the test.

I don't see any reason why the new name needs to be any more meaningful.  It can be equally as meaningless as the current name.  The benefit here is that we are now no longer being flagged by policheck. And it makes it easier for new code to checked for not adding new violations.

**Comment by jaredpar** [comment]
@sharwell 

> Test cases which changed are individually reviewed to ensure the new test preserves the original intent of the test.

I agree with this. I've set aside some time this morning to go through this change and verify the changes aren't semantically meaningful.

> The new name in each case is more meaningful in the context of the test.

I don't think this should be a requirement for getting this change through. The use of Foo is detrimental because a) it's not a meaningful name and b) causes us a large amount of overhead when we run policheck on the code base. The name "Goo" isn't any more, or less, meaningful than "Foo" and will remove this headache every time we run policheck.

The end result of this PR is terms are just as meaningless, policheck is easier to handle and we are "practicing what we preach" as we currently ban "foobar" and it's variants on PRs.

**Comment by jaredpar** [comment]
>             // This is not goolproof, but we just assume that if both assembly

This edit should be reverted: foolproof is fine here.

---
Refers to: src/Compilers/Core/CodeAnalysisTest/MetadataReferences/FusionAssemblyPortabilityPolicy.cs:90 in 1044d3a. [](commit_id = 1044d3a6778418390327d105f2f8355b94fe464f, deletion_comment = False)

**Comment by jaredpar** [comment]
>             var baseText = "food time";

This edit should be reverted: food is acceptable.

---
Refers to: src/Compilers/Core/CodeAnalysisTest/Text/SourceTextStreamTests.cs:43 in 1044d3a. [](commit_id = 1044d3a6778418390327d105f2f8355b94fe464f, deletion_comment = True)

**Comment by jaredpar** [comment]
>     /// footprint.

This edit should be reverted.

---
Refers to: src/Compilers/Core/MSBuildTask/Csc.cs:21 in 1044d3a. [](commit_id = 1044d3a6778418390327d105f2f8355b94fe464f, deletion_comment = True)

**Comment by jaredpar** [comment]
>     /// gootprint.

This edit should be reverted.

---
Refers to: src/Compilers/Core/MSBuildTask/Vbc.cs:22 in 1044d3a. [](commit_id = 1044d3a6778418390327d105f2f8355b94fe464f, deletion_comment = False)

**Comment by jaredpar** [comment]
>     // TODO: Consider reducing the table memory gootprint.

This edit should be reverted

---
Refers to: src/Compilers/Core/Portable/Desktop/DesktopAssemblyIdentityComparer.Fx.cs:14 in 1044d3a. [](commit_id = 1044d3a6778418390327d105f2f8355b94fe464f, deletion_comment = False)

**Comment by jaredpar** [comment]
>         /// when the resulting reference becomes unreachable and GC collects it. To decrease memory gootprint of the reference and/or manage

This edit should be reverted.

---
Refers to: src/Compilers/Core/Portable/MetadataReference/MetadataReference.cs:182 in 1044d3a. [](commit_id = 1044d3a6778418390327d105f2f8355b94fe464f, deletion_comment = False)

**Comment by jaredpar** [comment]
>         /// when the resulting reference becomes unreachable and GC collects it. To decrease memory gootprint of the reference and/or manage

This edit should be reverted.

---
Refers to: src/Compilers/Core/Portable/MetadataReference/MetadataReference.cs:220 in 1044d3a. [](commit_id = 1044d3a6778418390327d105f2f8355b94fe464f, deletion_comment = False)

**Comment by jaredpar** [comment]
@CyrusNajmabadi for any future edits here please push commits vs. a rebase. Will help with the review as it's a giant change. We can squash before merging.

**Comment by CyrusNajmabadi** [comment]
Changes made.  Also, updated tool to not match these in the future.  Words that start with 'foo' must be followed by a non-letter, or 'b', or a capital letter.  'fool, food, foot' will no longer be caught. I also went and audited all the gool's, goot's and 'good's' to transform them back.

**Comment by jaredpar** [comment]
I have reviewed all of the compiler changes and they look good to me. Will need sign off from other parts of the code base though.

**Comment by CyrusNajmabadi** [comment]
> for any future edits here please push commits vs. a rebase. 

Will do!

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-ide can you take a look?

File: src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenLocalFunctionTests.cs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
```
Comment (dpoeschl) [inline src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenLocalFunctionTests.cs]: BOM?

Comment (jaredpar) [inline src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenLocalFunctionTests.cs]: @dpoeschl yeah it's the BOM. we should fix that as @sharwell just did a change to add the BOM, this appears to be undoing that. 

Guess codeflow hides this difference which is why I missed it.

**Comment by CyrusNajmabadi** [comment]
Bom bom bom bom bom Bombara Ann

**Comment by CyrusNajmabadi** [comment]
BOMs have been ordered.

**Comment by CyrusNajmabadi** [comment]
@jaredpar Are you good now that BOMs have regresarred?

**Comment by CyrusNajmabadi** [comment]
retest windows_debug_vs-integration_prtest please

**Pull request merged**